pretix_mollie/payment.py

```diff
@@ -591,12 +591,19 @@ class MollieBanktransfer(MollieMethod):
     def requires_invoice_immediately(self):
         return self.settings.get('method_banktransfer_invoice_immediately', False, as_type=bool)
 
-    @transaction.atomic()
     def execute_payment(self, request: HttpRequest, payment: OrderPayment, retry=True):
-        p_orig = payment
-        if retry:
-            payment = OrderPayment.objects.select_for_update(of=OF_SELF).get(pk=payment.pk)
-        super().execute_payment(request, payment, retry)
+        err = None
+        with transaction.atomic():
+            p_orig = payment
+            if retry:
+                payment = OrderPayment.objects.select_for_update(of=OF_SELF).get(pk=payment.pk)
+            try:
+                super().execute_payment(request, payment, retry)
```

Comment:
I am not sure, what this PR tries to accomplish, but isn’t the problem that we want the token-refresh to not be dependent on a transaction-rollback due to payment errors? As the the token-refresh happens in `MollieMethod@execute_payment`, which is executed inside this `transaction.atomic`-block, it would still be rolled back as well?

Reply:
Ah, sorry, it actually works as intended